js/main.js

```diff
@@ -61,78 +61,80 @@ if (ACTIVE_ENV !== '') {
   use_envs = JSON.parse(localStorage.getItem('envIDs')) || ['main'];
 }
 
-// TODO: Move some of this to smaller components and/or use something like redux
-// to move state out of the app to a standalone store.
-class App extends React.Component {
-  state = {
-    connected: false,
-    readonly: false,
-    sessionID: null,
+function App() {
+  // -------------- //
+  // state varibles //
+  // -------------- //
+
+  // internal variables
+  const mounted = useRef(false);
+  const [connected, setConnected] = useState(false);
+  const [sessionInfo, setSessionInfo] = useState({ id: null, readonly: false });
+  const [resizeClickHappened, setResizeClickHappened] = useState(false);
+  const windowSize = useRef({
+    // width: 1280,
+    // cols: 100,
+  });
```

Comment:
It's a bit odd to initialize this ref with empty values, as it's unclear where these will eventually come from.

Reply:
Adapted & amended first commit. (I found it cleaner that way)